# Outlier Detection With IQR

This notebook uses the IQR method to identify outliers in a simple dataset.

In [1]:
import pandas as pd

scores = [
    ['Student 1', 10],
    ['Student 2', 12],
    ['Student 3', 15],
    ['Student 4', 18],
    ['Student 5', 20],
    ['Student 6', 22],
    ['Student 7', 25],
    ['Student 8', 27],
    ['Student 9', 30],
    ['Student 10', 32],
    ['Student 11', 35],
    ['Student 12', 100]
]

df = pd.DataFrame(scores, columns=['Student', 'Hours Studied'])

df

,Student,Hours Studied
0,Student 1,10
1,Student 2,12
2,Student 3,15
3,Student 4,18
4,Student 5,20
5,Student 6,22
6,Student 7,25
7,Student 8,27
8,Student 9,30
9,Student 10,32


In [2]:
Q1 = df['Hours Studied'].quantile(0.25)
Q3 = df['Hours Studied'].quantile(0.75)

IQR = Q3 - Q1

lower_limit = Q1 - 1.5 * IQR
upper_limit = Q3 + 1.5 * IQR

print(Q1)
print(Q3)
print(IQR)
print(lower_limit)
print(upper_limit)

17.25
30.5
13.25
-2.625
50.375


In [3]:
outliers = df.query('`Hours Studied` < @lower_limit or `Hours Studied` > @upper_limit')

outliers

,Student,Hours Studied
11,Student 12,100


## Results

I used the IQR method to identify outliers.

- Q1 = 17.25
- Q3 = 30.5
- IQR = 13.25
- Lower limit = -2.625
- Upper limit = 50.375

The value 100 is above the upper limit, so it is identified as an outlier.

## Interpretation

The outlier makes sense in the context of the data. Most students study between 10 and 35 hours per week, while one student studies 100 hours per week. This value is much higher than the other values and is above the IQR upper limit.

Therefore, 100 hours per week is an outlier according to the IQR method and is also unusually high compared with the rest of the dataset.